pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -719,18 +713,19 @@ public int getConcurrency() throws SQLException {
 
     int col = i - 1;
     int oid = fields[col].getOID();
-    if (oid != Oid.TIMESTAMP) {
-      throw new PSQLException(
-              GT.tr("Cannot convert the column of type {0} to requested type {1}.",
-                  Oid.toString(oid), "timestamp"),
-              PSQLState.DATA_TYPE_MISMATCH);
-    }
-    if (isBinary(i)) {
-      return getTimestampUtils().toLocalDateTimeBin(value);
+
+    if (oid == Oid.TIMESTAMP) {
+      if (isBinary(i)) {
+        return getTimestampUtils().toLocalDateTimeBin(value);
+      } else {
+        return getTimestampUtils().toLocalDateTime(castNonNull(getString(i)));
+      }
     }
 
-    String string = castNonNull(getString(i));
-    return getTimestampUtils().toLocalDateTime(string);
+    throw new PSQLException(
+        GT.tr("Cannot convert the column of type {0} to requested type {1}.",
+            Oid.toString(oid), "java.time.LocalDateTime"),
+        PSQLState.DATA_TYPE_MISMATCH);
```

Comment:
I would slightly disagree with moving throw at the end of the method.
If we require `Oid.TIMESTAMP`, then we just throw at the very beginning, so it is clear what are the valid input values, and we reduce the nesting of the subsequent code.

Reply:
I can check this again. I made the code here consistent with the other similar methods where the logic to check is more complicated.

For this part I can reverse it, but for the other methods where more than one type is expected, this would require to duplicate checks.

I think this is a matter of code style, I prefer it like that because its easier readable.